c7n/resources/ebs.py

```diff
@@ -210,6 +211,45 @@ def get_instance_mapping(self, resources):
         return {i['InstanceId']: i for i in instances}
 
 
+@filters.register('kms-alias')
+class KmsKeyAlias(ValueFilter):
+
+    schema = type_schema('kms-alias', rinherit=ValueFilter.schema)
+
+    def get_kms_aliases(self):
+        from c7n.resources.kms import KeyAlias
+        manager = KeyAlias(self.manager.ctx, {})
+        return [k for k in manager.resources()]
+
+    def process(self, resources, event=None):
+
+        key_aliases = self.get_kms_aliases()
+
+        def _user_kms_alias(resource):
+            for alias in key_aliases:
+                if 'TargetKeyId' in alias:
+                    if 'KmsKeyId' in resource:
+                        if alias['TargetKeyId'] in resource['KmsKeyId']:
+                            resource['Alias'] = alias
+                            break
+                    else:
+                        break
+
+        with self.executor_factory(max_workers=2) as w:
```

Comment:
since your not inlining any queries on the inner method, the executor factory becomes superflous (and actually slower).